### ISSUE TITLE
Refactoring: Move the ConfigFile class out of libsync

### ISSUE THREAD

**Issue by ogoffart** [owncloud/client#6213]
ConfigFile is currently in the libsync folder.
However, the sync engine should not read directly from the config file but should have all its parameter passed in the SyncOptions class.

Rationale: We don't want the tests, owncloudcmd, or other applications to share or mess with the same config file.

**Comment by guruz** [comment]
Only place missing:
```
src/libsync/creds/httpcredentials.cpp:    auto settings = ConfigFile::settingsWithGroup(Theme::instance()->appName());
```

**Comment by ckamm** [comment]
So it looks like we need to move the parts of `HttpCredentials` that interact with the keychain into gui and we're done. I don't think it's critical for 2.6, moving.

**Comment by ogoffart** [comment]
Right, we will refactor the credentials once we can get rid of shibboleth
